ddsrecorder_participants/src/cpp/recorder/mcap/McapHandler.cpp

```diff
@@ -223,6 +224,26 @@ void McapHandler::add_data(
                   );
     }
 
+    if (mcap_file_size_ + VERSION_METADATA_SIZE > configuration_.mcap_output_settings.max_file_size)
+    {
+        lock.unlock();
+
+        if (configuration_.mcap_output_settings.file_rotation)
+        {
+            logInfo(DDSRECORDER_MCAP_HANDLER, "Max file size reached, closing file and opening a new one...");
+            stop();
```

Comment:
I am not really sure if doing start and stop of the McapHandler is the best method to close and open the files

Reply:
After the redesign, there is no longer the need to `start` and `stop`.